### PR TITLE
Python 3.7 macOS Support

### LINKED ISSUE #1455: mxnet < 1.4.0 brings in an unsupported requests version
Current mxnet versions (prior to 1.4) bring in a version of requests that is unsupported and has reported vulnerabilities. We should upgrade our max dependency to 1.4.0 to fix this.

### PR DESCRIPTION
**Pull request by TobyRoseman**
I believe this is almost all of the changes we need (in this repository) to support Python 3.7. The only other changes that I know will be needed are in`.gitlab-ci.yml`.

Currently calling `predict(...)` on a coremltools model is broken in Python 3. This is being tracked in [an issue](https://github.com/apple/coremltools/issues/372) in the coremltools repository. We'll need to wait for that to get fixed and for a new version to be released.

All non-CoreML related unit tests pass for Python 3.7 on macOS.

This also fixes #1455

### CODE REVIEW

**Comment by znation** [comment]
Awesome, this is looking great! I can help with the `.gitlab-ci.yml` changes.

File: README.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ System Requirements
 
 Turi Create requires:
 
-* Python 2.7, 3.5, 3.6
+* Python 2.7, 3.5, 3.6, 3.7
```
Comment (nickjong) [inline README.md]: Where are we documenting the interaction with macOS/Linux?

File: src/python/turicreate/toolkits/object_detector/object_detector.py [hunk 2]
```diff
@@ -1164,8 +1165,14 @@ def _create_coreml_model(self, include_non_maximum_suppression,
...
         assert (self._model[23].name == 'pool5' and
                 self._model[24].name == 'specialcrop5')
-        del net._children[24]
-        net._children[23] = op
+
+        try:
+            del net._children[24]
+            net._children[23] = op
+        except KeyError:
+            # Newer versions of MXNet use string keys
```
Comment (nickjong) [inline src/python/turicreate/toolkits/object_detector/object_detector.py]: Ah, this definitely explains some of the issues we've seen reported here

File: README.md (hunk 1 above)
Comment (TobyRoseman) [reply README.md]: We're not yet. I'll add that.

Comment (TobyRoseman) [reply README.md]: I've add this.

**Comment by TobyRoseman** [comment]
Internal tests pass (minus known issues). I have cleaned/squash git commit history/messages. 

@znation @nickjong - I believe these changes are ready to go in now.

File: src/python/README.rst [hunk 3]
```diff
@@ -17,7 +17,7 @@ Turi Create supports:
 System Requirements
 -------------------
 
--  Python 2.7, 3.5, or 3.6
+-  Python 2.7, 3.5, 3.6, or 3.7 (macOS only)
```
Comment (znation) [inline src/python/README.rst]: We might want to clarify that Linux is supported on 2.7, 3.5, and 3.6? To me it looks like all of them are macOS only the way it's written.

File: src/unsupported_python/setup.py [hunk 4]
```diff
@@ -25,7 +25,7 @@ def run(self):
 
         * Linux x86_64 (including WSL on Windows 10).
         * macOS 10.12+ x86_64.
-        * Python 2.7, 3.5, or 3.6.
+        * Python 2.7, 3.5, 3.6, or 3.7 (macOS only).
```
Comment (znation) [inline src/unsupported_python/setup.py]: Same comment as above.

**Pull request merged**